core/src/main/java/hudson/util/ProcessKillingVeto.java

```diff
@@ -47,7 +49,7 @@
  * 
  * @author <a href="mailto:daniel.weber.dev@gmail.com">Daniel Weber</a>
  */
-public abstract class ProcessKillingVeto implements ExtensionPoint {
+public abstract class ProcessKillingVeto implements ExtensionPoint, Serializable {
```

Comment:
I am afraid it cannot work in such way. There may be already non-serializable extension point implementations. I would also vote against propagating new extension points to agents. I will likely have to write a wrapper class like `OnNodeProcessKillingVetoHandler` and return it from a method if the extension point supports vetoing there.